### DISCUSSION TITLE
What is encrypted at rest versus plaintext?

### DISCUSSION THREAD

**Discussion by wwwehr** [NillionNetwork/discussions#14] [Q&A] [answered]
We have several logical entities used on the Nillion Network. Can you tell me if any of these Nillion primitives are stored in plaintext or observable by the compute nodes?

1. ~Secrets~ - of course
2. Permissions - eg. Alice grants Bob permission to read her Secret. Is this instruction encrypted as well?
3. Programs - eg. Alice uploads a program and Bob executes it while acting as a compute node. Can Bob observe the program source/binary code?

Thanks!

**Comment by lumasepa** [comment, marked as answer]
2. Is not encrypted
3. Is not encrypted

They could be encrypted at rest but the node needs to be able to decrypt them to enforce the permissions and execute the programs
